src/StateSpace.hpp

```diff
@@ -28,6 +28,38 @@ class StateSpace {
     virtual T intermediateState(const T &source, const T &target, float stepSize) const = 0;
 
     /**
+     * An overloaded version designed for use in adaptive stepsize control.
+     *
+     * @param ascLimit The maximum size that the stepsize can be, even when there are no nearby obstacles
+     */
+    virtual T intermediateState(const T &source, const T &target, float prevStepSize, float ascGrowthRate, float defaultStepSize) const = 0;
+
+    /**
+     * Sets the aggresiveness of successful iterations of adaptive stepsize control.
+     *
+     * @param stepSizeScale How large the stepsize is scaled upon a successful iteration
+     */
+    virtual void setASCScale(float stepSizeScale) = 0;
```

Comment:
Does this determine the amount the step size increases each time or the max size it will reach?